docs/content/sips/00x-spin-build.md

```diff
@@ -0,0 +1,46 @@
+title = "SIP xxx - The `spin build` command"
+template = "main"
+date = "2022-04-22T14:53:30Z"
+---
+
+Summary: A Spin command for building components locally.
+
+Owner: radu.matei@fermyon.com
+
+Created: April 22, 2022
+
+Updated: April 22, 2022
+
+## Background
+
+A Spin application is made up of one or more components. When developing a
+multi-component application, it is very common to have multiple directories with
+source code for components — and when making a changes to components, having to
+manually go into the each component directory, compile the component, then go
+back to the directory with `spin.toml` can be a very repetitive task.
+
+## Proposal
+
+This SIP proposes a new top level Spin command that would execute the `command`
+field on the component configuration in a local `spin.toml` manifest file:
+
+```toml
+[[component]]
+id = "hello"
+source = "target/wasm32-wasi/release/spinhelloworld.wasm"
+
+[[component.build]]
+command = [ "cargo", "build", "--release", "--target", "wasm32-wasi" ]
+```
```

Comment:
I think this represents a good starting point for the MVP.

Reply:
I wonder if `build` could be an alternative kind of component source. Then `spin up` could check for the existence of e.g. `component.build.output` and run the build if necessary.

Reply:
Could also have a `rebuild-if-changed = "src/lib.rs"`

Reply:
At that point we can add an option to email the author on build failures, and by Zawinski's law we'll be done.

Kidding aside, let's not try to fit some partial implementation of `make` into Spin.  I like `spin build` as a convenient alternative to running manual build commands across multiple components.  Beyond that, trust each component's build tooling do its thing.

Reply:
Fair enough, though I think it would be worthwhile to "bless" (via docs, examples) a particular set of tools that make a good dev experience.